apps/dashboard/src/components/workflow-editor/steps/configure-step-form.tsx

```diff
@@ -0,0 +1,220 @@
+import { ConfirmationModal } from '@/components/confirmation-modal';
+import { PageMeta } from '@/components/page-meta';
+import { Button } from '@/components/primitives/button';
+import { Separator } from '@/components/primitives/separator';
+import { SidebarContent, SidebarFooter, SidebarHeader } from '@/components/side-navigation/sidebar';
+import TruncatedText from '@/components/truncated-text';
+import { EXCLUDED_EDITOR_TYPES } from '@/utils/constants';
+import { buildRoute, ROUTES } from '@/utils/routes';
+import {
+  IEnvironment,
+  StepDataDto,
+  StepTypeEnum,
+  UpdateWorkflowDto,
+  WorkflowOriginEnum,
+  WorkflowResponseDto,
+} from '@novu/shared';
+import { motion } from 'framer-motion';
+import { useMemo, useState } from 'react';
+import { RiArrowLeftSLine, RiArrowRightSLine, RiCloseFill, RiDeleteBin2Line, RiPencilRuler2Fill } from 'react-icons/ri';
+import { Link, useNavigate } from 'react-router-dom';
+import { SdkBanner } from '@/components/workflow-editor/steps/sdk-banner';
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from '@/components/primitives/form/form';
+import { CopyButton } from '@/components/primitives/copy-button';
+import { InputField, Input } from '@/components/primitives/input';
+import { useForm } from 'react-hook-form';
+import { useFormAutosave } from '@/hooks/use-form-autosave';
+import { getFirstBodyErrorMessage, getFirstControlsErrorMessage } from '@/components/workflow-editor/step-utils';
+import { ConfigureInAppStepTemplateCta } from '@/components/workflow-editor/steps/in-app/configure-in-app-step-template-cta';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { configureStepSchema } from '@/components/workflow-editor/schema';
+
+type ConfigureStepFormProps = {
+  workflow: WorkflowResponseDto;
+  environment: IEnvironment;
+  step: StepDataDto;
+  debouncedUpdate: (data: UpdateWorkflowDto) => void;
```

Comment:
I am not a big fan of this prop duplication. I'd pass a `save` prop that is debounced by default and accepts an option for immediate invocation. For example `save(newWorkflowData, { immediate: true })`. `

The key is to have one `save` across all form elements that are debounced by default, but you can also opt-out. The immediate should be part of the debounce mechanism so that subsequent immediate invocations still invoke the same function, and there is a single chain of immediate and debounced calls. 

See [here](https://lodash.com/docs/4.17.15#debounce).

Reply:
I think this works as expected right now. `update` and `debouncedUpdate` both call the same debounced func. `patch` is used only for the active field and it makes sense to be a different func.